contrib/platform/src/com/sun/jna/platform/mac/CoreFoundation.java

```diff
@@ -569,5 +628,5 @@ CFMutableDictionaryRef CFDictionaryCreateMutable(CFAllocatorRef alloc, int capac
      *            The {@code CFData} object to examine.
      * @return A read-only pointer to the bytes associated with {@code theData}.
      */
-    PointerByReference CFDataGetBytePtr(CFTypeRef theData);
+    PointerByReference CFDataGetBytePtr(CFDataRef theData);
```

Comment:
This looks strange. According to https://developer.apple.com/documentation/corefoundation/1543330-cfdatagetbyteptr?language=objc the return is 

> This function is guaranteed to return a pointer to a CFData object's internal bytes. CFData, unlike CFString, does not hide its internal storage

So this is a plain Pointer. The unittests `testCFData` works because it uses the wrong accessor. A `PointerByReference` is a pointer to a pointer sized buffer.  With `getPointer` you access the pointer to the buffer, with `getValue` you access the pointer contained in the buffer. As a general rule: If you use a `PointerByReference` and don't use its `getValue` function, something is fishy.

Reply:
Thanks -- if you only knew how much I googled "Pointer vs. PointerType vs. PointerByReference" and how many core dumps I generated until I got it right.  I think I have it sorted now and agree.